src/beqs.cpp

```diff
@@ -107,6 +109,20 @@ Rcpp::DataFrame processResponseEvent(Event event, const bool verbose) {
                     chk[i] = true;
                 }
             }
+            all_chk = all_chk and chk[i];
+        }
+
+        if(all_chk){
+            break;
+        }
+    }
+
+    if(!all_chk){               // Check if any columns have not been checked successfully - these will all set to fallback NA
+        for (int i=0; i<cols; i++) {
+            if(!chk[i]){
+                lst[i] = Rcpp::CharacterVector(rows, R_NaString);
```

Comment:
Which even I understood once I got to the third cup'o'joe  ;-)

Thanks for the enhancement!